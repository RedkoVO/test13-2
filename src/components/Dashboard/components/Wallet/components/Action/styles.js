import GraphLogo from '../../../../../../assets/images/svg/graph.svg'
import SendLogo from '../../../../../../assets/images/svg/sendIcon.svg'

export default theme => ({
  root: {
    width: '870px',
    height: '256px',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingTop: '90px',

    [theme.breakpoints.down('lg')]: {
      width: '100%',
      flexWrap: 'wrap'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%'
    },
    [theme.breakpoints.down('xs')]: {
      height: '100%'
    }
  },
  actionsStatistics: {
    width: '260px',
    height: '160px',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',

    [theme.breakpoints.down('lg')]: {
      width: '230px',
      height: '175px',
      marginTop: '15px'
    }
  },
  statisticsGraph: {
    width: '170px',
    height: '117px',

    '& img': {
      width: '100%',
      height: '100%'
    }
  },
  form: {
    textAlign: 'center'
  },
  statisticsTitle: {
    position: 'relative',
    fontSize: '29px',
    fontWeight: 700,
    color: '#5e7cff',
    textAlign: 'center',
    marginBottom: '25px',

    '&::before': {
      position: 'absolute',
      content: `""`,
      width: '16px',
      height: '16px',
      backgroundImage: `url(${GraphLogo})`,
      bottom: '8px',
      left: '-25px',
      backgroundRepeat: 'no-repeat',
      backgroundSize: '30px',
      backgroundPosition: 'center'
    }
  },
  actionsSend: {
    width: '260px',
    height: '226px',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',

    [theme.breakpoints.down('lg')]: {
      width: '230px',
      height: '175px'
    }
  },
  sendTitle: {
    position: 'relative',
    color: '#616161',
    fontFamily: 'ubBold',
    fontSize: '39px',
    textAlign: 'center',

    '&::before': {
      position: 'absolute',
      content: `""`,
      width: '17px',
      height: '17px',
      backgroundImage: `url(${SendLogo})`,
      bottom: '6px',
      left: '-25px',
      backgroundRepeat: 'no-repeat',
      backgroundSize: '16px',
      backgroundPosition: 'center'
    }
  },
  wrSendField: {
    display: 'flex',
    alignItems: 'flex-start',
    flexDirection: 'column'
  },
  sendAddressField: {
    width: '215px',
    height: '30px',
    boxSizing: 'border-box',
    border: 'none',
    backgroundColor: '#dddddd',
    outline: 'none',
    padding: '10px 20px',
    margin: 0,
    marginTop: '7px',
    borderRadius: '15px',
    fontSize: '9px',

    '&.input-error': {
      border: '1px solid red'
    }
  },
  sendAmmountField: {
    width: '210px',
    height: '30px',
    boxSizing: 'border-box',
    border: 'none',
    backgroundColor: '#dddddd',
    outline: 'none',
    padding: '10px 20px',
    margin: 0,
    marginTop: '7px',
    borderRadius: '15px',
    fontSize: '9px',

    '&.input-error': {
      border: '1px solid red'
    }
  },
  invoiceAmmountField: {
    width: '210px',
    height: '30px',
    boxSizing: 'border-box',
    border: 'none',
    backgroundColor: '#dddddd',
    outline: 'none',
    padding: '10px 20px',
    margin: 0,
    marginTop: '7px',
    borderRadius: '15px',
    fontSize: '9px'
  },
  sendBtn: {
    fontSize: '12px',
    fontWeight: 700,
    textDecoration: 'none',
    color: '#f0f0f0',
    backgroundColor: '#5e7cff',
    padding: '9px 20px',
    borderRadius: '20px',
    marginTop: '10px',

    '&.disabled': {
      opacity: '0.5',
      cursor: 'default'
    },

    '&:focus': {
      outline: 'none'
    }
  },
  walletTitle: {
    fontSize: '39px',
    fontWeight: 400,

    '& p': {
      margin: 0,
      fontFamily: 'ubBold'
    }
  },
  copyIcon: {
    marginLeft: '10px',
    width: '17px',
    height: '17px',
    cursor: 'pointer'
  },
  walletAddress: {
    display: 'flex',
    alignItems: 'center',
    fontSize: '11px',
    fontFamily: 'ubBold',
    padding: '10px 0',
    color: '#616161',
    textAlign: 'center'
  }
})
