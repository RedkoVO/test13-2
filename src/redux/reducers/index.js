import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import { APP_RESET } from '../actions/types'
import auth from './auth'
import balance from './balance'
import config from './config'
import history from './history'

const appReducer = combineReducers({
  form,
  auth,
  balance,
  config,
  history
})

export default (state, action) => {
  if (action.type === APP_RESET) {
    state = undefined
  }

  return appReducer(state, action)
}
