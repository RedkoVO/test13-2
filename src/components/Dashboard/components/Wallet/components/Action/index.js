import React from 'react'
import PropTypes from 'prop-types'
import { CopyToClipboard } from 'react-copy-to-clipboard'
import { Field, Form } from 'redux-form'
import withStyles from '@material-ui/core/styles/withStyles'
import cn from 'classnames'

import InputField from '../../../../../App/components/Form/InputField'

import CopyIcon from '../../../../../../assets/images/svg/copy.svg'

import styles from './styles'

const Action = ({
  classes,
  onSubmit,
  isDisabledButton,
  addressKey,
  onCopyKey
}) => (
  <div className={classes.root}>
    <div className={classes.actionsSend}>
      <Form className={classes.form} onSubmit={onSubmit}>
        <span className={classes.sendTitle}>Send now</span>
        <div className={classes.wrSendField}>
          <Field
            name="addressKey"
            type="text"
            className={classes.sendAddressField}
            component={InputField}
            placeholder="Put address you want to send the money"
          />
        </div>
        <div className={classes.wrSendField}>
          <Field
            name="amount"
            type="text"
            className={classes.sendAmmountField}
            component={InputField}
            placeholder="Set ammount"
          />
        </div>
        <button
          type="submit"
          className={cn(classes.sendBtn, { disabled: isDisabledButton })}
        >
          Send money
        </button>
      </Form>
    </div>

    <div className={classes.walletTitle}>
      <p>My Address:</p>
      <div className={classes.walletAddress}>
        {addressKey}
        <CopyToClipboard text={addressKey} onCopy={() => onCopyKey()}>
          <img className={classes.copyIcon} src={CopyIcon} alt="" />
        </CopyToClipboard>
      </div>
    </div>
  </div>
)

Action.propTypes = {
  classes: PropTypes.object,
  addressKey: PropTypes.string,
  onCopyKey: PropTypes.func,
  isDisabledButton: PropTypes.bool
}

export default withStyles(styles)(Action)
