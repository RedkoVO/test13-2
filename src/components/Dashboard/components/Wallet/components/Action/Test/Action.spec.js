import * as React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'

import Action from '../'

const Wrapper = reduxForm({ form: 'test' })(({ children }) => children)
const store = createStore(a => a, {})

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Wrapper>
        <Action
          onSubmit={() => { }}
          isDisabledButton={false}
        />
      </Wrapper>
    </Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})