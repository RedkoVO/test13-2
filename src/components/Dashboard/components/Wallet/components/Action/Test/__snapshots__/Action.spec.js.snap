// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Action-root-1"
>
  <div
    className="Action-actionsSend-6"
  >
    <form
      className="Action-form-4"
      onSubmit={[Function]}
    >
      <span
        className="Action-sendTitle-7"
      >
        Send now
      </span>
      <div
        className="Action-wrSendField-8"
      >
        <label
          htmlFor="addressKey"
        >
          Address:
        </label>
        <input
          className="Action-sendAddressField-9"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Put address you want to send the money"
          type="text"
          value=""
        />
      </div>
      <div
        className="Action-wrSendField-8"
      >
        <label
          htmlFor="amount"
        >
          Amount:
        </label>
        <input
          className="Action-sendAmmountField-10"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Set ammount"
          type="text"
          value=""
        />
      </div>
      <button
        className="Action-sendBtn-12"
        type="submit"
      >
        Send money
      </button>
    </form>
  </div>
  <div
    className="Action-actionsInvoice-14"
  >
    <span
      className="Action-invoiceTitle-15"
    >
      Invoice
    </span>
    <div
      className="Action-wrSendField-8"
    >
      <label
        htmlFor="invoice"
      >
        Email:
      </label>
      <input
        className="Action-invoiceAmmountField-11"
        name="invoice"
        placeholder="Set ammount you need to receive"
        type="text"
      />
    </div>
    <button
      className="Action-sendBtn-12"
      type="submit"
    >
      Send Invoice
    </button>
  </div>
</div>
`;
